libfreerdp/core/transport.c

```diff
@@ -891,6 +993,52 @@ int transport_write(rdpTransport* transport, wStream* s)
 	}
 
 	LeaveCriticalSection(&(transport->WriteLock));
+fail:
+	if (!length)
+		return writtenlength;
+	return status;
+}
+
+void transport_register_default_io_callbacks(rdpUpdate* update)
+{
+	rdpIoUpdate *io = calloc(1, sizeof(rdpIoUpdate));
+	io->context = update->context;
+	io->DataHandler = transport_io_data_handler;
+	io->Read = transport_io_data_read;
+	io->Write = transport_io_data_write;
+	freerdp_set_transport_callbacks(update->context, io);
+}
+
+
+void freerdp_set_transport_callbacks(rdpContext* context, void *io_callbacks)
+{
+	rdpIoUpdate* io = ((rdpIoUpdate*)io_callbacks);
```

Comment:
Never assume external stuff was allocated properly (nor by the same allocator functions as the library)

Reply:
despite this, the struct is allocated in the `_new` function, so doing just a `*context->update->io = *io_callbacks` will work just fine

Reply:
ok.